pkg/controllers/infracluster/aws.go

```diff
@@ -84,14 +124,91 @@ func (r *InfraClusterController) ensureAWSCluster(ctx context.Context, log logr.
 				Kind: awsv1.ControllerIdentityKind,
 				Name: "default",
 			},
+			// Set control plane load balancer configuration extracted from MAPI machines
+			ControlPlaneLoadBalancer:          controlPlaneLoadBalancer,
+			SecondaryControlPlaneLoadBalancer: secondaryControlPlaneLoadBalancer,
 		},
 	}
 
-	if err := r.Create(ctx, target); err != nil {
-		return nil, fmt.Errorf("failed to create InfraCluster: %w", err)
+	return target, nil
+}
+
+func (r *InfraClusterController) getAWSMAPIProviderSpec(ctx context.Context, cl client.Client) (*mapiv1beta1.AWSMachineProviderConfig, error) {
+	rawProviderSpec, err := r.getRawMAPIProviderSpec(ctx, cl)
+	if err != nil {
+		return nil, fmt.Errorf("unable to obtain MAPI ProviderSpec: %w", err)
 	}
 
-	log.Info(fmt.Sprintf("InfraCluster '%s/%s' successfully created", defaultCAPINamespace, r.Infra.Status.InfrastructureName))
+	providerSpec := &mapiv1beta1.AWSMachineProviderConfig{}
+	if err := yaml.Unmarshal(rawProviderSpec, providerSpec); err != nil {
+		return nil, fmt.Errorf("unable to unmarshal MAPI ProviderSpec: %w", err)
+	}
 
-	return target, nil
+	return providerSpec, nil
+}
+
+// extractLoadBalancerConfigFromMAPIProviderSpec extracts 1-2 control plane load balancers from a MAPI machine's provider spec.
+// The first load balancer is treated as the internal load balancer, and the second (if present) as the secondary one.
```

Comment:
This is backwards, at least from what I intended when I wrote the code upstream. CAPA only made external LBs, and the goal was to allow users to create internal ones as a secondary option.

Did you find that OpenShift sets the first as internal and the second as internal? It's very possible that the installer does this, because the names chosen upstream don't rely convey the public/private split well. As long as we're consistent with how they're used, I don't think it matters too much.

Reply:
Makes sense that if we have only 1 it's internal.

Since that's what OpenShift does and we're not really changing meanings from upstream fields, we're good to leave them as-is in this PR. The internal consistency in OpenShift is more important.

Also think Christian's comment should be incorporated.